helix-term/src/ui/editor.rs

```diff
@@ -569,24 +569,43 @@ impl EditorView {
         if let Some(syntax) = doc.syntax() {
             let text = doc.text().slice(..);
             use helix_core::match_brackets;
-            let pos = doc.selection(view.id).primary().cursor(text);
 
-            let pos = match_brackets::find_matching_bracket(syntax, doc.text(), pos)
-                .and_then(|pos| view.screen_coords_at_pos(doc, text, pos));
+            let default_style = Style::default()
+                .add_modifier(Modifier::REVERSED)
+                .add_modifier(Modifier::DIM);
 
-            if let Some(pos) = pos {
-                // ensure col is on screen
-                if (pos.col as u16) < viewport.width + view.offset.col as u16
-                    && pos.col >= view.offset.col
-                {
-                    let style = theme.try_get("ui.cursor.match").unwrap_or_else(|| {
-                        Style::default()
-                            .add_modifier(Modifier::REVERSED)
-                            .add_modifier(Modifier::DIM)
-                    });
+            let style_primary = theme
+                .try_get("ui.cursor.match.primary")
+                .unwrap_or(default_style);
+
+            let style_secondary = theme
+                .try_get("ui.cursor.match.secondary")
+                .unwrap_or(default_style);
```

Comment:
Instead of falling back to the default_style I think we should not have any fallback behavior for `ui.cursor.match.secondary` so that you can _not_ highlight secondary cursor matches

Reply:
Not highlighting the secondary cursor matches is still possible with `"ui.cursor.match.secondary" = {}`. So either way both are possible, the question is what should be the default. Personally I think highlighting secondary matches is a sensible default behavior, but it's up to you :)